numba/misc/dummyarray.py

```diff
@@ -314,6 +314,27 @@ def reshape(self, *newdims, **kws):
         if order not in 'CFA':
             raise ValueError('order not C|F|A')
 
+        # check for exactly one instance of -1 in newdims
+        unknownidx = -1
+        knownsize = 1
+        for i, dim in enumerate(newdims):
+            if dim < 0:
+                if unknownidx == -1:
+                    unknownidx = i
+                else:
+                    raise ValueError("can only specify one unknown dimension")
```

Comment:
Would be good to test that this branch is reached via an appropriate invalid scenario. Numba test suite uses this pattern for asserting exceptions are raised and correct.
https://github.com/numba/numba/blob/d7b1965cb8b5d13a92abd9b8f72f7da8c0fb7354/numba/tests/test_unicode.py#L1677-L1679